src/parse/asp/grammar_parse.go

```diff
@@ -227,6 +227,15 @@ func (p *parser) parseFuncDef() *FuncDef {
 		}
 	}
 	p.next(')')
+
+	if tok := p.l.Peek(); tok.Value != ":" {
+		p.next('-')
+		p.next('>')
+
+		tok := p.oneofval("[", "bool", "str", "int", "list", "dict", "function", "config", "buildrule")
```

Comment:
remove `"["` and `"buildrule"` (we don't have a separate type for build rules).

Reply:
oops!